Convert packaging/* to py3/py2.4 syntax

mscherer opened the pull request:
##### ISSUE TYPE
- Feature Pull Request

mscherer committed:
Convert packaging/* to py3/py2.4 syntax

gregdek commented:
Thanks @mscherer for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

abadger commented on packaging/os/rhn_register.py:
```diff
@@ -119,7 +119,8 @@
     import up2date_client.config
-except ImportError, e:
+except ImportError:
+    e = get_exception()
```
I think we may have to move the ansible.module_utils imports up above this otherwise get_exception won't be defined yet.  Could convert the ansible.module_utils imports to not be wildcard imports at the same time.

mscherer replied:
Yeah, I was wondering if it would work or not when I wrote the patch, but didn't dig much. Another solution is just to not get the exception in a variable and just show a simple generic error message.

abadger commented:
One comment for a thing that I think will throw an error if not changed a little bit.  I'm also going to be merging the travis.yml change from a whitelist to a blacklist soon https://github.com/ansible/ansible-modules-core/pull/3698  so you can convert the travis portion of htis change to using that as well.

mscherer commented:
Since sivel fixed the rhn_register code in https://github.com/ansible/ansible-modules-core/commit/1d87213de17e94c793e2850203aada5c339d850b,  the issue of order is also fixed, so I went ahead and rebased on top of the blacklist changes.

abadger commented:
Thanks @mscherer!  merged.
